query/src/servers/http/clickhouse_handler.rs

```diff
@@ -207,15 +303,46 @@ pub async fn clickhouse_handler_post(
         .map_err(BadRequest)?;
 
     let mut sql = params.query();
-    sql.push_str(body.into_string().await?.as_str());
-    let (plan, format) = PlanParser::parse_with_format(ctx.clone(), &sql)
-        .await
-        .map_err(BadRequest)?;
 
-    ctx.attach_query_str(&sql);
-    execute(ctx, plan, format, None, params)
-        .await
-        .map_err(InternalServerError)
+    let (stmts, _) = DfParser::parse_sql(sql.as_str(), ctx.get_current_session().get_type())
+        .unwrap_or_else(|_| (vec![], vec![]));
+
+    let settings = ctx.get_settings();
+    if settings.get_enable_new_processor_framework().unwrap() != 0
+        && !ctx.get_config().query.management_mode
+        && ctx.get_cluster().is_empty()
+        && settings.get_enable_planner_v2().unwrap() != 0
+        && !stmts.is_empty()
+        && stmts.get(0).map_or(false, InterpreterFactoryV2::check)
+    {
+        sql.push_str(body.into_string().await?.as_str());
+        let mut planner = Planner::new(ctx.clone());
+        let (plan, _) = planner.plan_sql(&sql).await.map_err(BadRequest)?;
+
+        let format = match plan.clone() {
+            Plan::Query {
+                s_expr: _,
+                metadata: _,
+                bind_context,
+            } => bind_context.format.clone(),
+            _ => None,
+        };
+
+        ctx.attach_query_str(&sql);
+        execute_v2(ctx, plan, format, None, params)
+            .await
+            .map_err(InternalServerError)
+    } else {
+        sql.push_str(body.into_string().await?.as_str());
```

Comment:
`sql.push_str(body.into_string().await?.as_str()) `  before if else?  otherwise it will lead to conflict with a pr I am working on.

Reply:
Ok.

Reply:
ignore me leave it as it is now,  my pr is small, I will wait for this pr merged and rebase myself.

Reply:
No problem. I have modified  with `unwrap()` together.